lib/limits.go

```diff
@@ -0,0 +1,13 @@
+// +build !windows
+
+package lib
+
+import "golang.org/x/sys/unix"
+
+// Getrlimit return the max file descriptors allocated by system
+// return the number of file descriptors max
+func Getrlimit() (uint64, error) {
```

Comment:
`Getrlimit` is only called from one package `agent/config`.  Until we need it in more places I think it would be good to keep it next to the one place that it is used. These two files can be moved to `agent/config/limits{_windows}.go` and `Getrlmit` can be unexported (`getrLimit`).